code/datums/components/storage/concrete/_concrete.dm

```diff
@@ -140,7 +140,17 @@
 	if(new_location)
 		//Reset the items values
 		_removal_reset(AM)
-		AM.forceMove(new_location)
+		if(!ismob(new_location))
+			AM.forceMove(new_location)
+		else // I hate this snowflake but "attempt_pickup()" calls forceMove() twice without this, sending item signal twice...
+			var/mob/taker = new_location
+			var/obj/item/target_item = AM
+			target_item.remove_outline()
+			target_item.add_fingerprint(taker)
+			// not silent && mob should stand on a turf && item is on a proper turf. 'get_turf(loc)' will be sent to the proc to show the animation
+			// if fails any of those, pickup animation will be skipped
+			if(!taker.put_in_active_hand(target_item, FALSE, !silent && isturf(new_location.loc) && get_turf(target_item.loc)))
+				taker.dropItemToGround(target_item) // does the same behaviour from attempt_pickup()
```

Comment:
This fundamentally loses a lot of the behaviour from attempt_pickup(). In reality, what we really want is either:
- Temporarilly remove from inventory doesn't call dropped() and is actually a temporary thing.
- Instead of temporarilly removing items from inventory, we can just check with a boolean value if items are allowed to be moved/picked up.

Reply:
> Temporarilly remove from inventory doesn't call dropped() and is actually a temporary thing.

Impossible, or highly not recommended. That breaks the core logic of `Moved()`.
The original logic already did `Temporarilly remove` and it made an issue that oldLoc and newLoc are not sane even if it doesn't send a duped signal.

> Instead of temporarilly removing items from inventory, we can just check with a boolean value if items are allowed to be moved/picked up.

ehh. I am not sure. I think I can just make a proc because these lines are the same anyway...?

Reply:
Hmm, no, I can't think any better logic, and I figured these lines merging into a proc is ugly, and becomes weird because of `silent` variable by storage.

Reply:
It could use transferItemToLoc() instead of temporarilyRemoveItemFromInventory() inside of attempt_pickup.

Reply:
Actually temporary removal should move the item to nullspace and not to a turf, which shouldn't call dropped multiple times. Moved() will be called multiple times, once to move to nullspace and once to move to the mob but that should be expected

Reply:
> It could use transferItemToLoc() instead of temporarilyRemoveItemFromInventory() inside of attempt_pickup.

I don't think any of `transferItemToLoc()` or `temporarilyRemoveItemFromInventory()` is necessary in `_concrete.dm`. That is a proc mainly for when you take off it from your equipment, not taking an item from storage.
I think we're not discussing about a thing correctly? Why do you think it's still temporary removal?

Reply:
I don't think its still temporary removal. I think it should use attempt_pickup() and not snowflake code, but attempt_pickup() should be made to use transferItemToLoc() so that the item isn't dropped to the ground and goes directly to the person instead.

Reply:
I made a new proc `move_to_mobhand()`. I think it fits?